standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/DatabaseProduct.java

```diff
@@ -83,6 +88,16 @@ protected DatabaseProduct() {}
   public static final String ORACLE_NAME = "oracle";
   public static final String UNDEFINED_NAME = "other";
 
+  public static DatabaseProduct determineDatabaseProduct(DataSource connPool,
+      Configuration conf) {
+    try (Connection conn = connPool.getConnection()) {
+      String s = conn.getMetaData().getDatabaseProductName();
+      return determineDatabaseProduct(s, conf);
+    } catch (SQLException e) {
+      throw new RuntimeException("Unable to get database product name", e);
```

Comment:
Is it possible to use some related Metastore runtime exception, instead of generic RuntimeException?

Reply:
if throwing other exceptions, we need to change our existing methods, such as the `ObjectStore.setConf` doesn't specify the allowed exceptions in `Configurable.setConf`.
Somewhere we catch and throw the RuntimeException upon the initialization. For example:
https://github.com/apache/hive/blob/master/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java#L412
https://github.com/apache/hive/blob/master/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java#L299

Reply:
can we throw `IllegalStateException`?

Reply:
@dengzhhu653, can we make this change? it's still a RuntimeException.

Reply:
sorry, somehow I missed this comment, fixed.